ng-dev/pr/config/index.ts

```diff
@@ -36,6 +36,13 @@ export interface PullRequestConfig {
   /** List of statuses that are required before a pull request can be merged. */
   requiredStatuses?: {type: 'check' | 'status'; name: string}[];
 
+  /**
+   * Whether pull requests should be merged using a conditional autosquash strategy.
+   * If a pull request contains fixup or squash commits, the autosquash strategy
+   * will be used. Otherwise, the Github API merge strategy will be used.
+   */
+  conditionalAutosquashMerge?: boolean;
```

Comment:
Would it make more sense to actually just add something like `rebase-with-fixups`  to the github API strategy? 

Or just change the `rebase` merge method to mean conditional autosquash and only have one "strategy"?